packages/beacon-node/src/metrics/server/http.ts

```diff
@@ -72,12 +73,17 @@ export async function getHttpMetricsServer(
 
   const activeSockets = new HttpActiveSocketsTracker(server, socketsMetrics);
 
-  const {port, address} = opts;
-  logger.info("Starting metrics HTTP server", {port, address: address ?? "127.0.0.1"});
-
   await new Promise<void>((resolve, reject) => {
-    server.once("error", reject);
-    server.listen(port, address, resolve);
+    server.once("error", (err) => {
+      logger.error("Error starting metrics HTTP server", opts, err);
+      reject(err);
+    });
+    server.listen(opts.port, opts.address, () => {
+      const {port, address: host, family} = server.address() as AddressInfo;
```

Comment:
should be safe to assume that this is a `AddressInfo` in our case
- `null` can't happen since we call it in listen callback
- `string` is only the case if it is a path to the pipe or socket but we always start server with address and port